### PR TITLE
ci: publish latest main to npm automatically

### PR DESCRIPTION
**Pull request by cajubelt**
trying for full release automation via github actions

### CODE REVIEW

**Comment by cajubelt** [comment]
we may also want [this plugin](https://github.com/semantic-release/git) to get the changelog and package.json updates pushed along with the release. I puzzled over it for a while and decided we can just try out the default plugins and see if we need to add it later.

**Comment by cajubelt** [comment]
I'm also not *totally certain* our branch protection rules won't block a release, but again, easiest way to determine that is probably just to test it

File: .github/workflows/workflow.yaml [hunk 1]
```diff
@@ -34,3 +34,29 @@ jobs:
       - run: yarn typecheck
       - run: yarn lint
       - run: yarn format:check
+  publish:
+    name: Publish to NPM
+    if: github.ref == 'refs/heads/main'
```
Comment (satish-ravi) [inline .github/workflows/workflow.yaml:39]: Does semantic-release ensure it doesn't run for a commit by semantic release or do we need a condition here?

**Review by satish-ravi — APPROVED** [review]

File: .github/workflows/workflow.yaml (hunk 1 above)
Comment (cajubelt) [reply .github/workflows/workflow.yaml:39]: it does

Comment (cajubelt) [reply .github/workflows/workflow.yaml:39]: puts [no-release] in the commit body I think

**Pull request merged**